File: delete all occurance of given key in dll.cpp
Node * deleteAllOccurrences(Node* head, int k) {
    // Write your code here      
      Node* temp=head;
        while(temp!=NULL){
            if(temp->data==k){
                if(temp == head){
                    head=temp->next;
                }
                Node *nextNode = temp->next;
                Node *prevNode = temp->prev;
                if(nextNode != NULL) nextNode->prev=prevNode;
                if(prevNode !=NULL) prevNode ->next= nextNode;
                
                free(temp);
                temp = nextNode;
            }
            else{
                temp=temp->next;
            }
        }
        return head;
}

File: delete all occurance of given key in dll.java

public class Solution {
    public static Node deleteAllOccurrences(Node head, int k) {
        // Write your code here.
         Node temp=head;
        while(temp!=null){
            if(temp.data==k){
                if(temp == head){
                    head=temp.next;
                }
                Node nextNode = temp.next;
                Node prevNode = temp.prev;
                if(nextNode != null) nextNode.prev=prevNode;
                if(prevNode !=null) prevNode.next= nextNode;
        
                temp = nextNode;
            }
            else{
                temp=temp.next;
            }
        }
        return head;
    }
}
